pkg/cloudinit/controlplane_init.go

```diff
@@ -53,6 +53,9 @@ func NewInitControlPlane(input InitControlPlaneInput) (CloudConfig, error) {
 
 	// run commands
 	config.RunCommands = append(config.RunCommands, input.PreRunCommands...)
+
+	config.RunCommands = append(config.RunCommands, "/capi/scripts/configure-snapstore-proxy.sh")
```

Comment:
Can we check if proxy is configured properly in the spec to before adding this script to `RunCommands`?